promql/engine.go

```diff
@@ -1173,7 +1173,8 @@ func (ev *evaluator) rangeEval(prepSeries func(labels.Labels, *EvalSeriesHelper)
 		}
 	}
 
-	for ts := ev.startTimestamp; ts <= ev.endTimestamp; ts += ev.interval {
+	for ts, step := ev.startTimestamp, -1; ts <= ev.endTimestamp; ts += ev.interval {
+		step++
```

Comment:
I found 'step' quite confusing.  Why is it initialized to -1 then immediately incremented, rather than initialized to 0 and incremented in the third part of the `for` ?

Reply:
Make sense.. i will do on the other places as well.